### ISSUE TITLE
Stubs broken with --strict-optional

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#1859]
Mypy complains about typeshed when using `--strict-optional`:

```
$ mypy --strict-optional t2.py
typeshed/stdlib/3/os/__init__.pyi:296: note: In module imported here,
t2.py:1: note: ... from here:
typeshed/stdlib/3/io.pyi: note: In class "BytesIO":
typeshed/stdlib/3/io.pyi:135: error: Signature of "write" incompatible with supertype "BinaryIO"
typeshed/stdlib/3/io.pyi:135: error: Return type of "write" incompatible with supertype "IO"
```

**Comment by gvanrossum** [comment]
Shouldn't this be a typeshed issue?

--Guido (mobile)

**Comment by JukkaL** [comment]
I added it here so that we don't lose track of it when preparing a release. I'll add a typeshed issue as well.

**Comment by gvanrossum** [comment]
Do you want this fixed before releasing 0.4.3?

**Comment by JukkaL** [comment]
Yeah, this would be nice to fix before the release since `os` is such a widely used module. I can look at this.

**Comment by gvanrossum** [comment]
Fixed by https://github.com/python/typeshed/pull/366

**Comment by gvanrossum** [comment]
(Actually, need a typeshed sync still.)

**Comment by gvanrossum** [comment]
Fixed by 80d7022b06c0b66840f0ee8d5f8bf98b372d1a46.
